tests/test_main_window.py

```diff
@@ -27,6 +27,44 @@
         )
 
 
+def test_exposure_changing(qtbot: "QtBot", main_window: MainWindow):
+    wait = 500
+    mmc = main_window._mmc
+    main_window.snap_channel_comboBox.setCurrentText("DAPI")
+    qtbot.wait_signal(main_window.snap_channel_comboBox.currentTextChanged)
+    qtbot.wait_signals([mmc.events.configGroupChanged, mmc.events.exposureChanged])
+
+    # Wait signals got this working on local but not remote?
+    # although it behaves fine as a user.
+    qtbot.wait(wait)
```

Comment:
any thoughts on how to make this work on remote? works from a user perspective but somehow not here?

Reply:
i think i fixed it. I had to use a dummy context manager on local in order to avoid this error:

```
xceptions caught in Qt event loop:
________________________________________________________________________________
Traceback (most recent call last):
  File "/home/ian/mambaforge/envs/pymm/lib/python3.9/site-packages/pytestqt/wait_signal.py", line 58, in _quit_loop_by_timeout
    self._cleanup()
  File "/home/ian/mambaforge/envs/pymm/lib/python3.9/site-packages/pytestqt/wait_signal.py", line 572, in _cleanup
    super()._cleanup()
  File "/home/ian/mambaforge/envs/pymm/lib/python3.9/site-packages/pytestqt/wait_signal.py", line 64, in _cleanup
    self._timeout_message = self._get_timeout_error_message()
  File "/home/ian/mambaforge/envs/pymm/lib/python3.9/site-packages/pytestqt/wait_signal.py", line 344, in _get_timeout_error_message
    error_message = self._get_expected_and_actual_signals_message()
  File "/home/ian/mambaforge/envs/pymm/lib/python3.9/site-packages/pytestqt/wait_signal.py", line 519, in _get_expected_and_actual_signals_message
    self._get_signal_string_representation_for_index(missing_signal_index)
  File "/home/ian/mambaforge/envs/pymm/lib/python3.9/site-packages/pytestqt/wait_signal.py", line 555, in _get_signal_string_representation_for_index
    signal_str_repr = self._signal_names[signal]
KeyError: <SignalInstance 'exposureChanged' on <pymmcore_plus.core._signals._CMMCoreSignaler object at 0x7fc9a9d3f9d0>>
________________________________________________________________________________
-------------------------------------------------------------------------- Captured log call --------------------------------------------------------------------------
ERROR    pymmcore_plus.core._mmcore_plus:_mmcore_plus.py:574 Exception occured in MMCorePlus callback 'exposureChanged': <SignalInstance 'exposureChanged' on <pymmcore_plus.core._signals._CMMCoreSignaler object at 0x7fc9a9d3f9d0>>

```